java/vector/src/test/java/org/apache/arrow/vector/TestValueVector.java

```diff
@@ -889,10 +890,94 @@ public void testNullableVarType2() {
    *  TODO:
    *
    *  The realloc() related tests below should be moved up and we need to
-   *  realloc related tests (edge cases) for more vector types.
+   *  add realloc related tests (edge cases) for more vector types.
    */
 
   @Test
+  public void testReallocAfterVectorTransfer() {
```

Comment:
Can you add tests for a NullableBitVector and NullableVarCharVector? Does MapVector or ListVector also need a test?